cirq-core/cirq/study/sweeps.py

```diff
@@ -34,10 +34,15 @@
 
 def _check_duplicate_keys(sweeps):
     keys = set()
+    key_count = 0
     for sweep in sweeps:
-        if any(key in keys for key in sweep.keys):
-            raise ValueError('duplicate keys')
+        key_count += len(sweep.keys)
         keys.update(sweep.keys)
+    # If the total length of the sweep keys
```

Comment:
This can be somewhat faster with a one-call set construction:
```
keys = set(itertools.chain.from_iterable(sweep.keys for sweep in sweeps))
key_count = sum(len(sweep.keys) for sweep in sweeps)
if key_count != len(keys):
   ...
```

Reply:
My tests show it as the same speed.  However, cleaner code and the same speed is still a win.  Done.